redux/features/prodInfo-slice.js

```diff
@@ -16,10 +22,23 @@ export const prodInfoSlice = createSlice({
       state.visibleImage = payload;
     },
     changeCurrentImage: (state, { payload }) => {
-      state.currentImage = payload;
+      if (typeof payload.isProd !== 'boolean') {
+        state.currentImage.url = payload.url;
+      } else if (!payload.url) {
+        state.currentImage.isProd = payload.isProd;
+      } else if (typeof payload.isProd === 'boolean' && payload.url) {
+        state.currentImage = payload;
+      }
+    },
```

Comment:
Edited changeCurrentImage to receive an object. If you pass in only the url prop with the obj it only update the url and vice versa with the isProp key. If you pass both the url and isProp key it will replace the entire object and update both.